OpenRA.Mods.Common/Projectiles/AreaBeam.cs

```diff
@@ -132,8 +136,38 @@ public AreaBeam(AreaBeamInfo info, ProjectileArgs args, Color color)
 			length = Math.Max((target - headPos).Length / speed.Length, 1);
 		}
 
+		void UpdateTarget()
+		{
+			if (args.GuidedTarget.IsValidFor(args.SourceActor))
+			{
+				var guidedTargetPos = args.GuidedTarget.CenterPosition;
+				var targetDistance = (guidedTargetPos - args.Source).Length;
+
+				// Only continue tracking target if it's within weapon range +
+				// 1 cell (1024) leniency to avoid edge case stuttering (start firing and immediately stop again).
+				if (targetDistance > args.Weapon.Range.Length + 1024)
+				{
+					continueTracking = false;
+					isTailTravelling = true;
+					return;
+				}
+
+				target = guidedTargetPos;
+			}
+
+			towardsTargetFacing = (target - args.Source).Yaw.Facing;
+
+			// Update the target position with the range we shoot beyond the target by
+			// I.e. we can deliberately overshoot, so aim for that position
+			var dir = new WVec(0, -1024, 0).Rotate(WRot.FromFacing(towardsTargetFacing));
+			target += dir * info.BeyondTargetRange.Length / 1024;
```

Comment:
if `!args.GuidedTarget.IsValidFor(args.SourceActor)` then this delta will be added to the already-offset position from the last tick.  I don't think this is what we want?